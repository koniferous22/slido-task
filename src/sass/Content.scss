@use 'base';

div#container section#content {
	@include base.page-section(50px, 80vw);
	@include base.fix-text-wrap;
	/*fixes overlaying fade from case list*/
	z-index: 1;
}
