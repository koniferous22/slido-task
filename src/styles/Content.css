section#content {
	text-align: center;
	max-width: 80vw;
	word-wrap: break-word;
	overflow-wrap: break-word;
	margin: 50px auto auto auto;
	/*fixes overlaying fade from case list*/
	z-index: 1;
}
