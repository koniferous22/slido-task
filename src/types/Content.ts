import { Case } from 'types/common/Case'

export interface ContentProps {
	displayedCase: Case | undefined;
	placeholder: string;
}
