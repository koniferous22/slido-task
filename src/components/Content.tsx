import React from 'react'

import { ContentProps } from 'types/Content'

import 'styles/Content.css'

const Content: React.FC<ContentProps> = ({placeholder, displayedCase}: ContentProps) => {
	return (
		<section id="content">
		{
			displayedCase ? (
					<>
						<h2 id="content-title">{displayedCase.title}</h2>
						<i>{displayedCase.date.toLocaleDateString()}</i>
						<p>{'Description: ' + displayedCase.description}</p>
					</>
				) : (
					<>
						{placeholder}
					</>
				)
		}
		</section>
	)
}

export default Content