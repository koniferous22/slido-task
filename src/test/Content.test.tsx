import React from 'react'
import renderer from 'react-test-renderer'
import Content from 'components/Content'

import { sampleCases } from 'test/data/sampleCases'

const placeholder = 'Sample Text'

it('Content placeholder renders correctly', () => {	
	const tree = renderer
		.create(
			<Content
				placeholder={placeholder}
				displayedCase={undefined}
			/>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('Content correctly', () => {	
	const tree = renderer
		.create(
			<Content
				placeholder={placeholder}
				displayedCase={sampleCases[2]}
			/>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});