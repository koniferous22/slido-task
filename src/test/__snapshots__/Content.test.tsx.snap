// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content correctly 1`] = `
<section
  id="content"
>
  <h2
    id="content-title"
  >
    I'm Rick Harrison, and this is my pawn shop.I work here with my old man and my son, Big Hoss. Everything in herehas a story and a price. One thing I've learned after 21 years –you never know WHAT is gonna come through that door.
  </h2>
  <i>
    2/26/2020
  </i>
  <p>
    Description: I'm Rick Harrison, and this is my pawn shop.I work here with my old man and my son, Big Hoss. Everything in herehas a story and a price. One thing I've learned after 21 years –you never know WHAT is gonna come through that door.
  </p>
</section>
`;

exports[`Content placeholder renders correctly 1`] = `
<section
  id="content"
>
  Sample Text
</section>
`;
